daml-model/daml/Tests/BankBICsTest.daml

```diff
@@ -8,14 +8,17 @@ module Tests.BankBICsTest where
 import Daml.Script
 
 import DA.Foldable
-import qualified DA.List as L
 import qualified DA.Set as S
 
 import Tests.PartyAlloc
 import Model.BankBIC
 
-createBankBICs banksWithBICs schedulers assemblers = do
-  forA_ banksWithBICs (\(banks, bic) -> submit (L.head $ S.toList banks) $ createCmd BankBIC with ..)
+createBankBICs : [(Party, Text)] -> S.Set Party -> S.Set Party -> Script ()
+createBankBICs banksWithBICs schedulerShards assemblerShards = do
+  forA_ banksWithBICs (\(bank, bic) -> submit bank $ createCmd BankBIC 
+    with 
+      bankShards = S.singleton bank
+      otherBanks = S.fromList $ filter (/= bank) (map fst banksWithBICs), ..)
```

Comment:
Isn't it better to create `allTheBanks : Set Party` and then just `delete` the current bank for every bank?

Reply:
Done